File: test_samples/java/clean.java
// Clean Java test file
// All functions used - no issues expected

public class UserService {
    public int add(int a, int b) {
        return a + b;
    }

    public static void main(String[] args) {
        UserService svc = new UserService();
        int result = svc.calculate(3, 4);
        System.out.println("Result: " + result);
    }

    public int calculate(int a, int b) {
        return add(a, b) * multiply(a, b);
    }

    public int multiply(int x, int y) {
        return x * y;
    }
}

File: test_samples/typescript/clean.ts
// Clean TypeScript test file - no issues expected

export function add(a: number, b: number): number {
    return a + b;
}

export function multiply(x: number, y: number): number {
    return x * y;
}

function main(): void {
    const result = calculate(3, 4);
    console.log('Result:', result);
}

function calculate(a: number, b: number): number {
    return add(a, b) * multiply(a, b);
}

main();
